### ISSUE TITLE
duration loss calculated in log domain or linear domain

### ISSUE THREAD

**Issue by MorganCZY** [janvainer/speedyspeech#19]
Could you explain the reasons for calculation duration loss in log domain, rather than linear domain directly?

**Comment by janvainer** [comment]
Hi, i tried using linear durations, but the problem was that the gradient was too large compared to the gradient in the rest of the network and the training dynamics was much worse

**Comment by wizardk** [comment]
> Hi, i tried using linear durations, but the problem was that the gradient was too large compared to the gradient in the rest of the network and the training dynamics was much worse

Hi janvainer, could you share the loss log in your experiment of training the duration extraction model, thanks a lot.

**Comment by wizardk** [comment]
> > Hi, i tried using linear durations, but the problem was that the gradient was too large compared to the gradient in the rest of the network and the training dynamics was much worse
> 
> Hi janvainer, could you share the loss log in your experiment of training the duration extraction model, thanks a lot.

I got training loss at 5.2865 and I'm not sure if there's a problem.

**Comment by janvainer** [comment]
Hi, could you describe your problem in more detail? Did you try to do some adjustments to the code?
Here is how the training went with ReduceLrOnPlateau and Noam schedulers:
![image](https://user-images.githubusercontent.com/31015638/115042448-c5473b80-9ed3-11eb-8a4c-88e2576f192e.png)

**Comment by wizardk** [comment]
Hi janvainer, thanks for your help.

Instead of using MinMaxNorm, I used a BatchNorm in the beginning of model.
Instead of using sample rate in 22Khz, I used sample rate in 16Khz.
I trimmed the start and end of silence in the wav files.

After I added MinMaxNorm, I got l1_loss about 0.04 and GuidedAttentionLoss about 0.0001 on the training set of LJSpeech-1.1.
Is the MinMaxNorm very important for training this model?

**Comment by wizardk** [comment]
Moreover, I trimmed the start and end of silence in the wav files.

**Comment by janvainer** [comment]
Hi, thanks for the uprades on your training! I found MinMax norm to be important for the teacher model, because a sigmoid activation is used on output and I got better results with it. The rest of the changes you made sound good to me. Let me know if the teacher model training works fine for you. :)

**Comment by wizardk** [comment]
> Hi, thanks for the uprades on your training! I found MinMax norm to be important for the teacher model, because a sigmoid activation is used on output and I got better results with it. The rest of the changes you made sound good to me. Let me know if the teacher model training works fine for you. :)

Hi janvainer, the training process was successful. Moreover, I used raw English characters rather than phonemes.

But when I extracted the durations, I found some big values (more than 100) in the middle and the tail of sentence. Some of them correspond to the English characters, some of them correspond to the full stop.
